boat_nav/src/navigator_node.py

```diff
@@ -219,33 +217,32 @@ def calc_global_max_vmg(wind_coming):
 def calc_cur_tack_max_vmg(wind_coming):
 
 	# TODO: Make a service and make more general for object avoidance
-	upper_bound = wind_coming + layline
-	lower_bound = wind_coming - layline
-	# Direct heading in irons on left of wind
-	if is_within_bounds(direct_heading, upper_bound, lower_bound):
-		theoretic_boat_speed = 0
+	upper_bound = angles.normalize(wind_coming + layline)
+	lower_bound = angles.normalize(wind_coming - layline)
 
-		if gtAngle(target_heading, wind_coming):
-			vmg_heading = (wind_coming + layline) % 360
-			theoretic_boat_speed = 2.5 * 0.514444
-		else :
-			vmg_heading = wind_coming - layline
-			if vmg_heading < 0:
-				vmg_heading += 360
-			theoretic_boat_speed = 2.5 * 0.514444
-
-	# Direct heading lies on navigatable path elsewhere in our current tack
-	elif (ltAngle(direct_heading, wind_coming) and ltAngle(target_heading, wind_coming)) or\
-		(gtAngle(direct_heading, wind_coming) and gtAngle(target_heading, wind_coming)):
-		theoretic_boat_speed = 2.5 * 0.514444 # 2.5 Knots to m/s (measured boat speed)
+	# If direct heading is in irons, then max_vmg will be on the edge of the no go zone
+	if angles.is_within_bounds(direct_heading, lower_bound, upper_bound):
+		theoretic_boat_speed = 2.5 * 0.514444 # 2.5 Knots to m/s
+
+		# Snap to whichever edge of the no go zone is closer
+		if angles.is_on_left(target_heading, wind_coming):
+			vmg_heading = upper_bound
+		else:
+			vmg_heading = lower_bound
+
+	# If the direct heading is somewhere in our current tack (direct_heading and target_heading on same side of wind)
+	elif (angles.is_on_left(direct_heading, wind_coming) and angles.is_on_left(target_heading, wind_coming)) or\
```

Comment:
Up to you, but this may be simpler if there was a `lie_on_same_side` utility.  May be more work than its worth but give it some thought.